app_dart/lib/src/request_handlers/scheduler/batch_backfiller.dart

```diff
@@ -99,15 +102,50 @@ class BatchBackfiller extends RequestHandler {
       });
       // Schedule all builds asynchronously.
       // Schedule after db updates to avoid duplicate scheduling when db update fails.
-      await Future.wait<void>(backfillRequestList(backfill));
+      await _scheduleWithRetries(backfill);
     } catch (error) {
       log.severe('Failed to update tasks when backfilling: $error');
     }
   }
 
+  /// Schedules tasks with retry when hitting pub/sub server errors.
+  Future<void> _scheduleWithRetries(List<Tuple<Target, FullTask, int>> backfill) async {
```

Comment:
Do you think this will have an impact on the secondary rate limit with github? I worry about us just retrying everything as we hit these limits regularly.

Reply:
This retry doesn't communicate with GitHub at all.
Though there is some check run logic in the scheduler, they are meant only for packages repo, which will not be called in the batch_backfiller api. This API handles only repo that is using batch scheduling logic, whereas packages is using `guaranteed` policy.
Additionally, this retries only tasks that fail to get scheduled. all happen in framework repo which will not touch the check run API.